docs/development/developer_guide.md

```diff
@@ -23,7 +23,7 @@ Install dependencies
 go mod tidy
 ```
 
-Build it
+Build the library
```

Comment:
I was also wondering what your thoughts are about making it clear that a Docker daemon needs to start running (with sufficient resources for CPU, memory and swap) before the build can begin.

I found that the `REGISTRY`, `TAG` and `CPU_ARCH` variables need to be set for the build to work but did not find that mentioned anywhere when I was first setting up the repo on my local.

```
make build REGISTRY=docker.io TAG=latest CPU_ARCH=arm64 # or other architecture
```

Also I found that I needed to log into nvcr.io using the instructions in https://stackoverflow.com/a/70970425/3528391. Otherwise the code failed due to not being to able to pull in the image in https://github.com/kubeflow/training-operator/blob/master/sdk/python/kubeflow/trainer/Dockerfile#L2.

Reply:
Do we need to run `make build` when using Kind @tenzen-y? I don't think we do, right? Maybe I have something wrong.

Reply:
I agree with @franciscojavierarceo since here it is focusing on `Building the operator`.